failed to build 1.4.9 on alpine

Seeing the following on alpine 20: 
```
In file included from /cloned/open62541/include/open62541/plugin/log.h:12,
                 from /cloned/open62541/plugins/include/open62541/plugin/log_stdout.h:10,
                 from /cloned/open62541/plugins/ua_log_stdout.c:8:
/cloned/open62541/build/src_generated/open62541/config.h:404:30: error: 'PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP' undeclared here (not in a function); did you mean 'PTHREAD_MUTEX_INITIALIZER'?
  404 | #define UA_LOCK_STATIC_INIT {PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP, 0}
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/cloned/open62541/plugins/ua_log_stdout.c:49:19: note: in expansion of macro 'UA_LOCK_STATIC_INIT'
   49 | UA_Lock logLock = UA_LOCK_STATIC_INIT;
```

--- a/include/open62541/config.h.in
+++ b/include/open62541/config.h.in
@@ -401,7 +401,11 @@ typedef struct {
     int mutexCounter;
 } UA_Lock;
 
+#if defined(__GLIBC__)
 #define UA_LOCK_STATIC_INIT {PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP, 0}
+#else
+#define UA_LOCK_STATIC_INIT {PTHREAD_MUTEX_INITIALIZER, 0}
+#endif
 
 static UA_INLINE void
 UA_LOCK_INIT(UA_Lock *lock) {

